include/boost/geometry/algorithms/detail/distance/multipoint_to_geometry.hpp

```diff
@@ -167,8 +217,27 @@ class multipoint_to_areal
                          boost::end(multipoint),
                          areal,
                          strategy);
+            bool is_ring = boost::is_same
+                    <
+                        typename tag<Areal>::type,
+                        ring_tag
+                    >::type::value;
+            bool is_multi_polygon = boost::is_same
+                    <
+                        typename tag<Areal>::type,
+                        multi_polygon_tag
+                    >::type::value;
+
+            if (!is_ring && !is_multi_polygon)
+            {
+                strategy::distance::services::swap<Strategy>::apply(res);
+            }
+            return res;
         }
-        return 0;
+        return_type res;
+        strategy::distance::services::result_set_unique_point<Strategy>
```

Comment:
Isn't it rather a `detail` instead of `service`? Or have you mimicked already existing designs? I don't remember exactly but AFAIR there are many convoluted implementation details in `distance`/`comparable_distance`. So I'd understand if that was simply using the existing way of doing things around this algorithm.

Reply:
Maybe I missing something on this topic whether sth is `detail` or `service`. The `detail` namespace is for algorithms right? The `services` is for strategies. As commented above I followed already existing designs for manipulations of the result.

Reply:
Ok I see there is detail namespace for strategies too but the main argument holds.